src/test/java/com/example/report/controller/ReportControllerTest.java

```diff
@@ -0,0 +1,148 @@
+        LocalDate to = LocalDate.of(2022, 12, 31);
+
+        String responseAsAString = mockMvc.perform(MockMvcRequestBuilders.post("/report/generate/{from}/{to}", from, to))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString();
+
+        List<ReportEntryDTO> actualReportEntryDTOsList = objectMapper.readValue(responseAsAString, new TypeReference<>() {
+        });
+
+        Assertions.assertEquals(392, actualReportEntryDTOsList.get(0).getNetAmount().intValue());
+        Assertions.assertEquals(98, actualReportEntryDTOsList.get(0).getPersonalIncomeTax().intValue());
+    }
+
+    @Test
+    void getAllReports() throws Exception {
+        LocalDate testDate = LocalDate.of(2022, 10, 26);
+        ReportDTO report1 = new ReportDTO(1L, testDate, testDate);
+        ReportDTO report2 = new ReportDTO(2L, testDate, testDate);
+
+        List<ReportDTO> expectedReportDTOS = List.of(report1, report2);
+        reportRepository.saveAll(reportMapper.dtoToEntity(expectedReportDTOS));
+
+        String responseAsAString = mockMvc.perform(MockMvcRequestBuilders.get("/report"))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString();
+        List<ReportDTO> actualReportDTOsList = objectMapper.readValue(responseAsAString, new TypeReference<>() {
+        });
+        assertThat(expectedReportDTOS).isEqualTo(actualReportDTOsList);
+    }
+
+    @Test
+    void getReportEntriesByReportId() throws Exception {
+        ReportEntry reportEntry = ReportEntry.builder()
+                .id(1L)
+                .report(new Report(8L, LocalDate.of(2022, 1, 30), LocalDate.of(2022, 1, 30)))
+                .employee(new Employee(1L, "Nika", "Avalishvili", "Department", "Position", "Email", true, true))
+                .benefit(new Benefit(1L, "Bonus", "Accrual", "Gross"))
+                .netAmount(BigDecimal.valueOf(6000))
+                .personalIncomeTax(BigDecimal.valueOf(1000))
+                .pensionsFund(BigDecimal.valueOf(500))
+                .grossAmount(BigDecimal.valueOf(7500))
+                .document(new Document(1L, LocalDate.of(2022, 1, 2), LocalDate.of(2022, 2, 2), 1L, 1L, BigDecimal.valueOf(600)))
+                .build();
+
+        Long reportId = reportEntryRepository.save(reportEntry).getReport().getId();
+
+        String responseAsAString = mockMvc.perform(MockMvcRequestBuilders.get("/report/{id}", reportId))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString();
+
+        List<ReportEntry> actualReportEntryDTOsList = objectMapper.readValue(responseAsAString, new TypeReference<>() {
+        });
+
+        assertThat(reportEntry.getNetAmount().intValue()).isEqualTo(actualReportEntryDTOsList.get(0).getNetAmount().intValue());
+        assertThat(reportEntry.getPensionsFund().intValue()).isEqualTo(actualReportEntryDTOsList.get(0).getPensionsFund().intValue());
```

Comment:
IntValue makes are assertions less accurate, so let's use bigDecimals themselves here and let's use `isEqualByComparingTo()`